x-pack/plugins/lens/public/visualizations/xy/annotations/actions/save_action.tsx

```diff
@@ -0,0 +1,281 @@
+
+    savedId = layer.annotationGroupId;
+  }
+
+  return { id: savedId, config: groupConfig };
+};
+
+/** @internal exported for testing only */
+export const onSave = async ({
+  state,
+  layer,
+  setState,
+  eventAnnotationService,
+  toasts,
+  modalOnSaveProps: { newTitle, newDescription, newTags, closeModal, newCopyOnSave },
+  dataViews,
+}: {
+  state: XYState;
+  layer: XYAnnotationLayerConfig;
+  setState: StateSetter<XYState, unknown>;
+  eventAnnotationService: EventAnnotationServiceType;
+  toasts: ToastsStart;
+  modalOnSaveProps: ModalOnSaveProps;
+  dataViews: DataViewsContract;
+}) => {
+  let savedInfo: Awaited<ReturnType<typeof saveAnnotationGroupToLibrary>>;
+  try {
+    savedInfo = await saveAnnotationGroupToLibrary(
+      layer,
+      { newTitle, newDescription, newTags, newCopyOnSave },
+      eventAnnotationService,
+      dataViews
+    );
+  } catch (err) {
+    toasts.addError(err, {
+      title: i18n.translate(
+        'xpack.lens.xyChart.annotations.saveAnnotationGroupToLibrary.errorToastTitle',
+        {
+          defaultMessage: 'Failed to save "{title}"',
+          values: {
+            title: newTitle,
+          },
+        }
+      ),
+    });
+
+    return;
+  }
+
+  const newLayer: XYByReferenceAnnotationLayerConfig = {
+    ...layer,
+    annotationGroupId: savedInfo.id,
+    __lastSaved: savedInfo.config,
+  };
+
+  setState({
+    ...state,
+    layers: state.layers.map((existingLayer) =>
+      existingLayer.layerId === newLayer.layerId ? newLayer : existingLayer
+    ),
+  });
+
+  closeModal();
+
+  toasts.addSuccess({
+    title: i18n.translate(
+      'xpack.lens.xyChart.annotations.saveAnnotationGroupToLibrary.successToastTitle',
+      {
+        defaultMessage: 'Saved "{title}"',
+        values: {
+          title: newTitle,
+        },
+      }
+    ),
+    text: ((element) =>
+      render(
+        <p>
+          <FormattedMessage
+            id="xpack.lens.xyChart.annotations.saveAnnotationGroupToLibrary.successToastBody"
+            defaultMessage="View or manage in the {link}"
+            values={{
+              link: <a href="#">annotation library</a>,
+            }}
+          />
+        </p>,
+        element
+      )) as MountPoint,
+  });
+};
+
+export const getSaveLayerAction = ({
+  state,
+  layer,
+  setState,
+  eventAnnotationService,
+  toasts,
+  savedObjectsTagging,
+  dataViews,
+}: {
+  state: XYState;
+  layer: XYAnnotationLayerConfig;
+  setState: StateSetter<XYState, unknown>;
+  eventAnnotationService: EventAnnotationServiceType;
+  toasts: ToastsStart;
+  savedObjectsTagging?: SavedObjectTaggingPluginStart;
+  dataViews: DataViewsContract;
+}): LayerAction => {
+  const neverSaved = !isByReferenceAnnotationsLayer(layer);
+
+  const displayName = i18n.translate(
+    'xpack.lens.xyChart.annotations.saveAnnotationGroupToLibrary',
+    {
+      defaultMessage: 'Save annotation group',
```

Comment:
Wording change.

```suggestion
      defaultMessage: 'Save to library',
```

Reply:
Addressed in https://github.com/elastic/kibana/pull/159057